src/main/java/org/mule/extension/oauth2/internal/AbstractGrantType.java

```diff
@@ -206,6 +209,36 @@ public void setMuleContext(MuleContext muleContext) {
     this.resolver = new DeferredExpressionResolver(muleContext.getExpressionManager());
   }
 
+
+  @Override
+  public boolean equals(Object obj) {
+    if (obj instanceof AbstractGrantType) {
+      AbstractGrantType other = (AbstractGrantType) obj;
+      return Objects.equals(clientId, other.clientId) &&
+          Objects.equals(clientSecret, other.clientSecret) &&
+          Objects.equals(scopes, other.scopes) &&
+          Objects.equals(tokenManager, other.tokenManager) &&
+          Objects.equals(tokenUrl, other.tokenUrl) &&
+          literalEquals(responseAccessToken, other.responseAccessToken) &&
+          literalEquals(responseRefreshToken, other.responseRefreshToken) &&
+          literalEquals(responseExpiresIn, other.responseExpiresIn) &&
+          Objects.equals(parameterExtractors, other.parameterExtractors) &&
+          literalEquals(refreshTokenWhen, other.refreshTokenWhen) &&
+          Objects.equals(tlsContextFactory, other.tlsContextFactory) &&
+          Objects.equals(proxyConfig, other.proxyConfig);
+    }
+
+    return false;
+  }
+
+  @Override
+  public int hashCode() {
+    return 31 * hash(
```

Comment:
Does the formatter like this? OMG

Reply:
it not only likes it, it changed what I did manually in favor of this beauty

Reply:
:wheaton: